### PR TITLE
Terraform support for Schematics Agents

### PR DESCRIPTION
**Pull request by VaishnaviGopal**
image
image
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: ibm/service/schematics/resource_ibm_schematics_agent.go [hunk 1]
```diff
@@ -0,0 +1,1314 @@
...
+			},
+			"system_state": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Computed state of the agent.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status_code": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
```
Comment (kavya498) [inline ibm/service/schematics/resource_ibm_schematics_agent.go:371]: Inner attributes of computed field should also be computed.
Please make sure it is followed in all schema attributes

File: website/docs/d/schematics_agent.html.markdown [hunk 2]
```diff
@@ -0,0 +1,163 @@
+---
+layout: "ibm"
+page_title: "IBM : ibm_schematics_agent"
+description: |-
+  Get information about schematics_agent
+subcategory: "Schematics Service API"
```
Comment (kavya498) [inline website/docs/d/schematics_agent.html.markdown:6]: existing resources has subcategory as Schematics
Please change `Schematics Service API` to `Schematics` in all docs

File: website/docs/r/schematics_agent.html.markdown [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+	* `updated_at` - (String) The agent prs job updation time.
+	* `updated_by` - (String) Email address of user who ran the agent prs job.
+* `system_state` - (List) Computed state of the agent.
+Nested scheme for **system_state**:
+	* `status_code` - (String) Agent Status.
+	  * Constraints: Allowable values are: `error`, `normal`, `in_progress`, `pending`, `draft`.
+	* `status_message` - (String) The agent status message.
+* `updated_at` - (String) The agent registration updation time.
+* `updated_by` - (String) Email address of user who updated the agent registration.
+
```
Comment (kavya498) [inline website/docs/r/schematics_agent.html.markdown:196]: Below mentioned sections are generic and is available in provider docs..
```
## Provider Configuration
### Static credentials
### Environment variables
```
kindly remove these sections from the docs

**Review by kavya498 — CHANGES_REQUESTED** [review]

File: website/docs/r/schematics_agent.html.markdown (hunk 3 above)
Comment (VaishnaviGopal) [reply website/docs/r/schematics_agent.html.markdown:196]: done

File: website/docs/d/schematics_agent.html.markdown (hunk 2 above)
Comment (VaishnaviGopal) [reply website/docs/d/schematics_agent.html.markdown]: done

File: ibm/service/schematics/resource_ibm_schematics_agent.go (hunk 1 above)
Comment (VaishnaviGopal) [reply ibm/service/schematics/resource_ibm_schematics_agent.go:371]: done

File: ibm/service/schematics/resource_ibm_schematics_agent.go [hunk 4]
```diff
@@ -0,0 +1,1290 @@
...
+			},
+		},
+	}
+}
+
+func ResourceIbmSchematicsAgentValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "schematics_location",
```
[the remark this answers is not in the document]
Comment (VaishnaviGopal) [reply ibm/service/schematics/resource_ibm_schematics_agent.go]: removed.

**Review by hkantare — APPROVED** [review]

**Pull request merged**